client/post-editor/editor-visibility/index.jsx

```diff
@@ -284,88 +311,149 @@ const EditorVisibility = React.createClass( {
 		}
 	},
 
-	render() {
-		const visibility = this.getVisibility();
+	renderPrivacyPopover( visibility ) {
 		const icons = {
 			password: 'lock',
 			'private': 'not-visible',
 			'public': 'visible'
 		};
+
+		return (
+			<Button
+					compact
+					borderless
+					onClick={ this.togglePopover }
+					ref="setVisibility"
+				>
+				<Gridicon icon={ icons[ visibility ] || 'visible' } /> { visibility }
+				<Popover
+					className="editor-visibility__popover"
+					isVisible={ this.state.showPopover }
+					onClose={ this.closePopover }
+					position={ 'bottom left' }
+					context={ this.refs && this.refs.setVisibility }
+				>
+					<div className="editor-visibility__dialog">
+						<FormFieldset className="editor-fieldset">
+							<FormLegend className="editor-fieldset__legend">
+								{ this.translate( 'Post Visibility' ) }
+								<Gridicon
+									icon="info-outline"
+									size={ 18 }
+									className={ classNames( 'editor-visibility__dialog-info', { is_active: this.state.showVisibilityInfotips } ) }
+									onClick={ this.toggleVisibilityInfotips }
+								/>
+							</FormLegend>
+							<FormLabel>
+								<FormRadio
+									name="site-visibility"
+									value="public"
+									onChange={ this.updateVisibility }
+									checked={ 'public' === visibility }
+								/>
+								<span>
+									{
+										this.props.isPrivateSite
+										? this.translate( 'Visible for members of the site', { context: 'Editor: Radio label to set post visibility' } )
+										: this.translate( 'Public', { context: 'Editor: Radio label to set post visible to public' } )
+									}
+								</span>
+							</FormLabel>
+							{ this.renderVisibilityTip( 'public' ) }
+
+							<FormLabel>
+								<FormRadio
+									name="site-visibility"
+									value="private"
+									onChange={ this.onSetToPrivate }
+									checked={ 'private' === visibility }
+								/>
+								<span>{ this.translate( 'Private', { context: 'Editor: Radio label to set post to private' } ) }</span>
+							</FormLabel>
+							{ this.renderVisibilityTip( 'private' ) }
+							<FormLabel>
+								<FormRadio
+									name="site-visibility"
+									value="password"
+									onChange={ this.updateVisibility }
+									checked={ 'password' === visibility }
+								/>
+								<span>{ this.translate( 'Password Protected', { context: 'Editor: Radio label to set post to password protected' } ) }</span>
+							</FormLabel>
+							{ this.renderVisibilityTip( 'password' ) }
+							{ visibility === 'password' ? this.renderPasswordInput() : null }
+						</FormFieldset>
+					</div>
+				</Popover>
+			</Button>
+		);
+	},
+
+	renderPrivacyDropdown( visibility ) {
+		const dropdownItems = [
+			{
+				label: this.translate( 'Publicly Viewable' ),
+				icon: 'visible',
+				value: 'public',
+				onClick: () => {
+					this.translate( 'Public', { context: 'Editor: Radio label to set post visible to public' };
```

Comment:
Good catch. Got sloppy when I updated the translation strings. Should be fixed now.